File: src/main/java/info/hieule/arx_automation/app/DemoPublisher.java
package info.hieule.arx_automation.app;

import com.rabbitmq.client.Channel;
import com.rabbitmq.client.Connection;
import com.rabbitmq.client.ConnectionFactory;

import java.io.IOException;
import java.util.concurrent.TimeoutException;

public class DemoPublisher {
    private final static String QUEUE_NAME = "datasets";

    public static void main(String[] args) throws IOException, TimeoutException {
        ConnectionFactory connectionFactory = new ConnectionFactory();
        connectionFactory.setHost("localhost");
        connectionFactory.setConnectionTimeout(3000);

        try (Connection connection = connectionFactory.newConnection();
             Channel channel = connection.createChannel()
        ) {
            channel.queueDeclare(QUEUE_NAME, false, false, false, null);

            String message = "Hello World!";

            channel.basicPublish("", QUEUE_NAME, null, message.getBytes());

            System.out.println(" [x] Sent '" + message + "'");
        }
    }
}

File: src/main/kotlin/info/hieule/arx_automation/app/DemoPublisher.kt
@file:JvmName("DemoPublisher")

package info.hieule.arx_automation.app

import com.rabbitmq.client.ConnectionFactory

private const val QUEUE_NAME = "dataset"

fun main(args: Array<String>) {
    val connectionFactory = ConnectionFactory()
    connectionFactory.host = "localhost"
    connectionFactory.connectionTimeout = 3000
    connectionFactory.newConnection().use { connection ->
        connection.createChannel().use { channel ->
            channel.queueDeclare(QUEUE_NAME, false, false, false, null)
            val message = "Hello World!"
            channel.basicPublish("", QUEUE_NAME, null, message.toByteArray())
            println(" [x] Sent '$message'")
        }
    }
}
